internal/compiler/langtype.rs

```diff
@@ -784,70 +784,55 @@ pub fn unit_product_length_conversion(
     a: &[(Unit, i8)],
     b: &[(Unit, i8)],
 ) -> Option<LengthConversionPowers> {
```

Comment:
This is what i had in mind as a simplification (still with only px and phx)

```rust
/// If the `Type::UnitProduct(a)` can be converted to `Type::UnitProduct(a)` by multiplying
/// by the scale factor, return that scale factor, otherwise, return None
pub fn unit_product_length_conversion(a: &[(Unit, i8)], b: &[(Unit, i8)]) -> Option<i8> {
    let mut units = [0i8; 16];
    for (u, count) in a {
        units[*u as usize] += count;
    }
    for (u, count) in b {
        units[*u as usize] -= count;
    }

    if units[Unit::Px as usize] != -units[Unit::Phx as usize] {
        return None;
    }

    let power = units[Unit::Phx as usize];
    units[Unit::Px as usize] = 0;
    units[Unit::Phx as usize] = 0;
    units.into_iter().all(|x| x == 0).then(|| power)
}
```
Tested to work.


When adding another unit, i guess it would be something like

```rust
    if units[Unit::Px as usize] + units[Unit::Phx as usize]  + units[Unit::Rem as usize] != 0 {
        return None;
    }
    let result = LengthConversionPowers { rem_to_px_power: units[Unit::Phx as usize], px_to_phx_power: units[Unit::Rem as usize]})
    //... = 0
    units.into_iter().all(|x| x == 0).then(|| result)

```

or something like that

Reply:
Nice - thanks!